nixops/evaluation.py

```diff
@@ -0,0 +1,204 @@
+        )
+    if not os.path.exists(expr_path):
+        expr_path = os.path.dirname(__file__) + "/../nix"
+    return expr_path
+
+
+def eval(
+    # eval-machine-info args
+    networkExpr: NetworkFile,  # Flake conditional
+    uuid: str,
+    deploymentName: str,
+    networkExprs: List[str] = [],
+    args: Dict[str, str] = {},
+    pluginNixExprs: List[str] = [],
+    checkConfigurationOptions: bool = True,
+    # Extend internal defaults
+    nix_path: List[str] = [],
+    # nix-instantiate args
+    nix_args: Dict[str, Any] = {},
+    attr: Optional[str] = None,
+    extra_flags: List[str] = [],
+    # Non-propagated args
+    stderr: Optional[TextIO] = None,
+) -> Any:
+
+    exprs: List[str] = list(networkExprs)
+    if not networkExpr.is_flake:
+        exprs.append(networkExpr.network)
+
+    argv: List[str] = (
+        ["nix-instantiate", "--eval-only", "--json", "--strict", "--show-trace"]
+        + [os.path.join(get_expr_path(), "eval-machine-info.nix")]
+        + ["-I", "nixops=" + get_expr_path()]
+        + [
+            "--arg",
+            "networkExprs",
+            py2nix([RawValue(x) if x[0] == "<" else x for x in exprs]),
+        ]
+        + [
+            "--arg",
+            "args",
+            py2nix({key: RawValue(val) for key, val in args.items()}, inline=True),
+        ]
+        + ["--argstr", "uuid", uuid]
+        + ["--argstr", "deploymentName", deploymentName]
+        + ["--arg", "pluginNixExprs", py2nix(pluginNixExprs)]
+        + ["--arg", "checkConfigurationOptions", json.dumps(checkConfigurationOptions)]
+        + list(itertools.chain(*[["-I", x] for x in (nix_path + pluginNixExprs)]))
+        + extra_flags
+    )
+
+    for k, v in nix_args.items():
+        argv.extend(["--arg", k, py2nix(v, inline=True)])
+
+    if attr:
+        argv.extend(["-A", attr])
+
+    if networkExpr.is_flake:
+        argv.extend(["--allowed-uris", get_expr_path()])
+        argv.extend(["--argstr", "flakeUri", networkExpr.network])
+
+    try:
+        ret = subprocess.check_output(argv, stderr=stderr, text=True)
+        return json.loads(ret)
+    except OSError as e:
+        raise Exception("unable to run ‘nix-instantiate’: {0}".format(e))
+    except subprocess.CalledProcessError:
+        raise NixEvalError
+
+
+def eval_network(nix_expr: NetworkFile) -> NetworkEval:
+    try:
+        result = eval(
+            networkExpr=nix_expr,
+            uuid="dummy",
+            deploymentName="dummy",
+            attr="info.network",
+        )
+    except Exception:
+        raise NixEvalError("No network attribute found")
+
+    if result.get("storage") is None:
+        raise MalformedNetworkError(
+            """
+WARNING: NixOps 1.0 -> 2.0 conversion step required
+
+NixOps 2.0 added support for multiple storage backends.
+
+Upgrade steps:
+1. Open %s
+2. Add:
+    network.storage.legacy = {
+      databasefile = "~/.nixops/deployments.nixops"
```

Comment:
There's a reference to `databasefile` here that doesn't seem to exist in the code